Shared/Samples/Edit with branch versioning/EditWithBranchVersioningView.swift

```diff
@@ -0,0 +1,250 @@
+                                .font(.callout)
+                        }
+                        editDamageTypeButton
+                    }
+                    .padding(5)
+                }
+                .onSingleTapGesture { screenPoint, mapPoint in
+                    if model.selectedFeature != nil && !model.onDefaultVersion {
+                        // Shows the move confirmation alert if there is already a feature selected.
+                        isMovingFeature = true
+                        moveLocation = mapPoint
+                    } else {
+                        // Identifies and selects the feature at the tap location.
+                        selectedAction = .selectFeature(screenPoint: screenPoint, mapPoint: mapPoint)
+                    }
+                }
+                .alert(
+                    "Confirm Move",
+                    isPresented: $isMovingFeature,
+                    presenting: moveLocation,
+                    actions: { mapPoint in
+                        Button("Cancel", role: .cancel, action: {})
+                        Button("Move") {
+                            model.selectedFeature?.geometry = mapPoint
+                            selectedAction = .updateFeature
+                        }
+                    },
+                    message: { _ in
+                        Text("Do you want to move the selected feature?")
+                    }
+                )
+                .task(id: selectedAction) {
+                    guard let selectedAction else { return }
+                    defer { self.selectedAction = nil }
+                    calloutPlacement = nil
+                    
+                    do {
+                        switch selectedAction {
+                        case .setUp:
+                            try await model.setUp()
+                            
+                        case .makeVersion:
+                            try await model.createVersion(parameters: versionParameters!)
+                            
+                        case .switchToVersion(let version):
+                            try await model.switchToVersion(named: version)
+                            
+                        case .updateFeature:
+                            try await model.updateFeature()
+                            
+                        case .selectFeature(let screenPoint, let mapPoint):
+                            model.clearSelection()
+                            
+                            // Identifies the feature on the feature layer at the tapped point.
+                            let result = try await mapViewProxy.identify(
+                                on: model.featureLayer,
+                                screenPoint: screenPoint,
+                                tolerance: 10
+                            )
+                            guard let feature = result.geoElements.first as? Feature else { return }
```

Comment:
A guard statement nested this deep really breaks the mental model for guard statements. Consider using an if statement instead.